lib/helpers.py

```diff
@@ -589,13 +610,14 @@ def _to_parquet(df, file_name):
         df.to_parquet(file_name, engine='pyarrow')
 
     @staticmethod
-    def _improve_types(df):
+    def _improve_types(df, export_user_ids):
         """
         Use more memory efficient types for ts,user_id and ab_test_group
         """
         df['ts'] = pd.to_datetime(df['ts'])
         df['ts'] = (df['ts'].astype('int64') / 1e9).astype('int32')
-        df['user_id'] = df['user_id'].apply(xxhash.xxh64_intdigest).astype('int64')
+        if not export_user_ids:
```

Comment:
**To discuss:**
I'm unsure about this implementation altogether. For the sake of a single operation, we sacrifice quite significant amount of RAM for the entire report.

Meanwhile, we could instead implement a sister function to `read_csv`, that would piece-wise read the CSVs and incrementally write user IDs for each group upon processing of a chunk to the destination file.

Thus, we would keep our report-related optimization, and have the necessary functionality, which can be run independently of all other report manipulations.

Reply:
Now I do it when reading csv and in chunks